File: mycollab-mobile/src/main/scala/com/mycollab/mobile/module/crm/events/ActivityEvent.scala
package com.mycollab.mobile.module.crm.events

import com.mycollab.events.ApplicationEvent

/**
  * @author MyCollab Ltd
  * @since 5.4.9
  */
object ActivityEvent {

  class TaskAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class TaskEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class TaskRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class MeetingAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class MeetingEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class MeetingRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class CallAdd(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class CallEdit(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class CallRead(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoList(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

  class GotoRelatedItems(source: AnyRef, data: AnyRef) extends ApplicationEvent(source, data) {}

}

File: mycollab-mobile/src/main/java/com/mycollab/mobile/module/crm/event/ActivityEvent.kt
package com.mycollab.mobile.module.crm.event

import com.mycollab.events.ApplicationEvent

/**
 * @author MyCollab Ltd
 * @since 6.0.0
 */
class ActivityEvent {
    class TaskAdd(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class TaskEdit(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class TaskRead(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class MeetingAdd(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class MeetingEdit(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class MeetingRead(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class CallAdd(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class CallEdit(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class CallRead(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class GotoList(source: Any, data: Any?) : ApplicationEvent(source, data) 

    class GotoRelatedItems(source: Any, data: Any?) : ApplicationEvent(source, data) 
}